src/common/string/compare.ts

```diff
@@ -11,3 +11,25 @@ export const compare = (a: string, b: string) => {
 
 export const caseInsensitiveCompare = (a: string, b: string) =>
   compare(a.toLowerCase(), b.toLowerCase());
+
+export const fuzzyCompare = (filter: string, word: string) => {
+  if (filter === "") {
+    return true;
+  }
+
+  for (let i = 0; i <= filter.length; i++) {
+    const letter = filter[0];
+    const pos = word.indexOf(letter);
+
+    if (pos < 0) {
+      return false;
+    }
+
+    const newWord = word.substring(pos + 1);
+    const newFilter = filter.substring(1);
+
+    return fuzzyCompare(newFilter, newWord);
+  }
+
+  return true;
```

Comment:
Had to add this due to eslint requirement. We should never get to this line, but in case we do, figured we should err on the side of `true` when filtering items. Better to show an entity mistakenly than to hide one unfairly?